### DISCUSSION TITLE
Best practice to use Cypress Studio with Gherkin/Cucumber ?

### DISCUSSION THREAD

**Discussion by StephaneTactill** [cypress-io/cypress#16446] [Plugins]
Hi,

I have setup my work project using cypress and the plugin cypress-cucumber-preprocessor (https://github.com/TheBrainFamily/cypress-cucumber-preprocessor).
I have also recently turned on Cypress Studio and I am wondering if there is a best practice about how to use both of these tools?

Right now my development process is like so:
1) I write in Gherkin the test to be done (a ".feature" file)
2) I create the .js file where I will describe the step-definitions
3) I start Cypress and start Cypress Studio in the .js file I created in 2)
4) I reproduce, manually, the steps written in 1) and the code is generated by Studio
5) I save the code generated
6) I go back to my IDE and I start formatting the code generated into a Gherkin format

This is tedious and not very future proof. I guess there must be a better way. If you know one please share.
Best to you.

**Comment by panzarino** [comment]
@StephaneTactill We're planning on adding the ability to copy generated commands rather than just saving them to a file which should alleviate a bit of your pain. As for Gherkin/cucumber direct support, that would have to be something you request from the plugin maintainer.
